observability/src/test/java/io/grpc/observability/LoggingTest.java

```diff
@@ -17,55 +17,117 @@
+import io.grpc.observability.interceptors.ConfigFilterHelper;
+import io.grpc.observability.interceptors.ConfigFilterHelper.MethodFilterParams;
 import io.grpc.observability.interceptors.InternalLoggingChannelInterceptor;
 import io.grpc.observability.interceptors.InternalLoggingServerInterceptor;
+import io.grpc.observability.interceptors.LogHelper;
 import io.grpc.observability.logging.GcpLogSink;
 import io.grpc.observability.logging.Sink;
 import io.grpc.testing.GrpcCleanupRule;
 import io.grpc.testing.protobuf.SimpleServiceGrpc;
 import java.io.IOException;
+import java.util.Map;
 import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
 
 @RunWith(JUnit4.class)
 public class LoggingTest {
 
   @Rule
   public final GrpcCleanupRule cleanupRule = new GrpcCleanupRule();
 
-  private static final String PROJECT_ID = "project-id";
+  private static final String PROJECT_ID = "PROJECT";
+  private static final Map<String, String> locationTags = ImmutableMap.of(
+      "project_id", "PROJECT",
+      "location", "us-central1-c",
+      "cluster_name", "grpc-observability-cluster",
+      "namespace_name", "default" ,
+      "pod_name", "app1-6c7c58f897-n92c5");
+  private static final Map<String, String> customTags = ImmutableMap.of(
+      "KEY1", "Value1",
+      "KEY2", "VALUE2");
 
   /**
    * Cloud logging test using LoggingChannelProvider and LoggingServerProvider.
    */
   @Ignore
   @Test
-  public void clientServer_interceptorCalled()
+  public void clientServer_interceptorCalled_logAlways()
       throws IOException {
     Sink sink = new GcpLogSink(PROJECT_ID);
+    LogHelper spyLogHelper = spy(new LogHelper(sink, TimeProvider.SYSTEM_TIME_PROVIDER,
+        locationTags, customTags));
+    ConfigFilterHelper mockFilterHelper = mock(ConfigFilterHelper.class);
+    MethodFilterParams logAlwaysFilterParams =
+        new MethodFilterParams(true, 0, 0);
+    when(mockFilterHelper.isMethodToBeLogged(anyString()))
+        .thenReturn(logAlwaysFilterParams);
     LoggingServerProvider.init(
-        new InternalLoggingServerInterceptor.FactoryImpl(sink, null, null, null));
+        new InternalLoggingServerInterceptor.FactoryImpl(spyLogHelper, mockFilterHelper));
     Server server = ServerBuilder.forPort(0).addService(new LoggingTestHelper.SimpleServiceImpl())
         .build().start();
     int port = cleanupRule.register(server).getPort();
     LoggingChannelProvider.init(
-        new InternalLoggingChannelInterceptor.FactoryImpl(sink, null, null, null));
+        new InternalLoggingChannelInterceptor.FactoryImpl(spyLogHelper, mockFilterHelper));
     ManagedChannel channel = ManagedChannelBuilder.forAddress("localhost", port)
         .usePlaintext().build();
     SimpleServiceGrpc.SimpleServiceBlockingStub stub = SimpleServiceGrpc.newBlockingStub(
         cleanupRule.register(channel));
```

Comment:
If channel is not used elsewhere just inline channel (expr on line 90)

Reply:
Done